example/src/main/java/com/graphhopper/example/RoutingExampleTC.java

```diff
@@ -8,55 +8,81 @@
 import com.graphhopper.config.CHProfile;
 import com.graphhopper.config.Profile;
 import com.graphhopper.reader.osm.GraphHopperOSM;
-import com.graphhopper.routing.util.CarFlagEncoder;
 import com.graphhopper.routing.util.EncodingManager;
-import com.graphhopper.routing.util.FlagEncoder;
-import com.graphhopper.util.Helper;
-import com.graphhopper.util.PMap;
+import com.graphhopper.util.Parameters;
 
 import java.util.Arrays;
 
 import static com.graphhopper.util.Parameters.Curbsides.CURBSIDE_ANY;
 import static com.graphhopper.util.Parameters.Curbsides.CURBSIDE_RIGHT;
 
 /**
- * Routing with turn costs
+ * Routing with turn costs. Also see {@link RoutingExample} for more details.
  */
 public class RoutingExampleTC {
     public static void main(String[] args) {
         String relDir = args.length == 1 ? args[0] : "";
         GraphHopper hopper = createGraphHopperInstance(relDir + "core/files/andorra.osm.pbf");
-        routing(hopper);
+        routeWithTurnCosts(hopper);
+        routeWithTurnCostsAndCurbsides(hopper);
+        routeWithTurnCostsAndOtherUTurnCosts(hopper);
     }
 
-    // see RoutingExample for more details
-    public static void routing(GraphHopper hopper) {
-        GHRequest req = new GHRequest(42.50822, 1.533966, 42.506899,1.525372).
+    public static void routeWithTurnCosts(GraphHopper hopper) {
+        GHRequest req = new GHRequest(42.50822, 1.533966, 42.506899, 1.525372).
+                setProfile("car");
+        route(hopper, req, 1038, 63_000);
+    }
+
+    public static void routeWithTurnCostsAndCurbsides(GraphHopper hopper) {
+        GHRequest req = new GHRequest(42.50822, 1.533966, 42.506899, 1.525372).
                 setCurbsides(Arrays.asList(CURBSIDE_ANY, CURBSIDE_RIGHT)).
                 setProfile("car");
-        GHResponse rsp = hopper.route(req);
+        route(hopper, req, 1730, 112_000);
+    }
 
+    public static void routeWithTurnCostsAndOtherUTurnCosts(GraphHopper hopper) {
+        GHRequest req = new GHRequest(42.50822, 1.533966, 42.506899, 1.525372).
+                setCurbsides(Arrays.asList(CURBSIDE_ANY, CURBSIDE_RIGHT)).
+                putHint(Parameters.CH.DISABLE, true).
```

Comment:
This is a bit ugly because we do not get an error if CH is still enabled. When we change the u-turn costs (and CH is enabled) this change is simply ignored without throwing an error or something.

Reply:
We can start to be more restrictive here and throw errors for curbside&noneCH? But this might break clients. Or we have a strict mode (default) and legacy mode ;)

Reply:
There is this todo here: https://github.com/graphhopper/graphhopper/blob/ff85f037d637e3813dc588b2105df6422afbfa7a/core/src/main/java/com/graphhopper/routing/Router.java#L230-L233

And yes I think we did not do this for some backward-compatibility reasons. Not sure what the exact implications are if we change this.

Reply:
> and throw errors for curbside&noneCH

curbsides+nonCH is not the problem. But what is not possible is changing the u-turn costs with CH.

> Or we have a strict mode (default) and legacy mode ;)

Hm this sounds complicated? Or do you mean some flag that controls this somehow (ignore or throw error).

Reply:
> curbsides+nonCH is not the problem. 

Ah, due to the request properties thought you mean the problem that this parameter combination does not work for node-based yet ...

> Or do you mean some flag that controls this somehow (ignore or throw error).

Yes. But not sure how ugly the code would get with that.

Reply:
Honestly, as long as their is no complaint I do not really feel like changing anything here :) Adding some ignore parameter makes it more complicated and maybe at some point we no longer need the backward-compatibility here?

Reply:
> Adding some ignore parameter makes it more complicated

Yes, I also do not like it.

> maybe at some point we no longer need the backward-compatibility here?

Unfortunately this goes not away the longer we wait. Instead it is more the opposite. Maybe we introduce restrictions step by step ... have no hard opinion here but it is tricky.

Reply:
We would have to figure out how many requests there are that use `u_turn_costs=<some value>` but do *not* use `ch.disable=true`. (and similar with `short_fastest.time/distance_factor`

Reply:
Ah and then there might be a problem with `ProfileResolver`, i.e. when `u_turn_costs` is given but no `profile` is specified. Before we had profiles we used `u_turn_costs` to determine which CH preparation shall be used in case there are multiple ones using the same vehicle/weighting, but different u_turn_costs.